src/main/java/org/jabref/gui/exporter/WriteMetadataToLinkedPdfsAction.java

```diff
@@ -0,0 +1,207 @@
+            this.xmpPreferences = xmpPreferences;
+            this.stateManager = stateManager;
+            this.dialogService = dialogService;
+
+            updateMessage(Localization.lang("Writing metadata..."));
+        }
+
+        @Override
+        protected Void call() throws Exception {
+            if (stateManager.getActiveDatabase().isEmpty()) {
+                return null;
+            }
+
+            for (int i = 0; i < entries.size(); i++) {
+                BibEntry entry = entries.get(i);
+                updateProgress(i, entries.size());
+
+                // Make a list of all PDFs linked from this entry:
+                List<Path> files = entry.getFiles().stream()
+                                        .map(file -> file.findIn(stateManager.getActiveDatabase().get(), filePreferences))
+                                        .filter(Optional::isPresent)
+                                        .map(Optional::get)
+                                        .filter(FileUtil::isPDFFile)
+                                        .toList();
+                if (files.isEmpty()) {
+                    LOGGER.debug("Skipped empty entry '{}'",
+                            entry.getCitationKey().orElse(entry.getAuthorTitleYear(16)));
+                    skipped++;
+                } else {
+                    for (Path file : files) {
+                        updateMessage(Localization.lang("Writing metadata to {}", file.getFileName()));
+
+                        if (Files.exists(file)) {
+                            try {
+                                WriteMetadataToSinglePdfAction.writeMetadataToFile(
+                                        file,
+                                        entry,
+                                        databaseContext,
+                                        abbreviationRepository,
+                                        entryTypesManager,
+                                        fieldPreferences,
+                                        filePreferences,
+                                        xmpPreferences);
+                                entriesChanged++;
+                            } catch (Exception e) {
+                                LOGGER.error("Error while writing XMP data to pdf '{}'", file, e);
+                                failedWrittenFiles.add(file);
+                                errors++;
+                            }
+                        } else {
+                            LOGGER.debug("Skipped non existing pdf '{}'", file);
+                            skipped++;
+                        }
+                    }
+                }
+                updateMessage(Localization.lang("Processing..."));
+            }
+
+            updateMessage(Localization.lang("Finished"));
+            dialogService.notify(Localization.lang("Finished writing metadata for database %0 (%1 succeeded, %2 skipped, %3 errors).",
```

Comment:
JabRef uses "library" for .bib files, not "database" as term.

```suggestion
            dialogService.notify(Localization.lang("Finished writing metadata for library %0 (%1 succeeded, %2 skipped, %3 errors).",
```